services/management/checks.go

```diff
@@ -212,24 +212,64 @@ func (s *ChecksAPIService) ListSecurityChecks(ctx context.Context, req *manageme
 		m[c] = struct{}{}
 	}
 
-	checks, err := s.checksService.GetChecks()
+	results, err := s.checksService.GetChecks()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get available checks list")
 	}
 
-	res := make([]*managementpb.SecurityCheck, 0, len(checks))
-	for _, c := range checks {
+	checks := make([]*managementpb.SecurityCheck, 0, len(results))
+	if req != nil && req.FilterParams != nil {
+		if req.FilterParams.Category != nil {
+			// we only support filtering by category for now.
+			// later, other "keys" can be supported by "AND-ing" the filters i.e., the response from each filter
+			// can be passed to the next.
+			if filterValues, ok := req.FilterParams.Category.Value.(*managementpb.InFilter_StringValues); ok {
+				checks = filterByCategory(results, m, filterValues.StringValues.Values)
+				return &managementpb.ListSecurityChecksResponse{Checks: checks}, nil
+			}
+
+			return nil, errors.New("no valid 'string_values' for category filter")
+		}
+	}
+
+	for _, c := range results {
 		_, disabled := m[c.Name]
-		res = append(res, &managementpb.SecurityCheck{
+		checks = append(checks, &managementpb.SecurityCheck{
 			Name:        c.Name,
 			Disabled:    disabled,
 			Summary:     c.Summary,
 			Description: c.Description,
 			Interval:    convertInterval(c.Interval),
+			Category:    c.Category,
 		})
 	}
 
-	return &managementpb.ListSecurityChecksResponse{Checks: res}, nil
+	return &managementpb.ListSecurityChecksResponse{Checks: checks}, nil
+}
+
+func filterByCategory(checks map[string]check.Check, disabledChecks map[string]struct{}, categories []string) []*managementpb.SecurityCheck {
+	res := make([]*managementpb.SecurityCheck, 0, len(checks))
+
+	cm := make(map[string]struct{}, len(categories))
+	for _, category := range categories {
+		cm[category] = struct{}{}
+	}
+
+	var disabled bool
+	for _, ch := range checks {
+		if _, ok := cm[ch.Category]; ok {
+			_, disabled = disabledChecks[ch.Name]
+			res = append(res, &managementpb.SecurityCheck{
+				Name:        ch.Name,
+				Disabled:    disabled,
+				Summary:     ch.Summary,
+				Description: ch.Description,
+				Interval:    convertInterval(ch.Interval),
+				Category:    ch.Category,
+			})
+		}
+	}
```

Comment:
looks like duplication of this and block above https://github.com/percona/pmm-managed/pull/1065/files#diff-ae29f66284415f440fd7f8446d2c0c11735f3807133a9b2d9889c9a1a40d0101R235-R245 can be simplified using just by adding `or`.
```suggestion
	for _, ch := range checks {
		if _, ok := cm[ch.Category]; ok || len(cm) == 0 {
			_, disabled = disabledChecks[ch.Name]
			res = append(res, &managementpb.SecurityCheck{
				Name:        ch.Name,
				Disabled:    disabled,
				Summary:     ch.Summary,
				Description: ch.Description,
				Interval:    convertInterval(ch.Interval),
				Category:    ch.Category,
			})
		}
	}
```

Reply:
Yeah, we can either skip it or call it always (whether there's a category filter or not) and both will work.

I think skipping makes it clearer which filter gets applied or not.

Reply:
In other endpoints we return all values in cases of empty filters.
Even if we skip the function we can simplify using the same code style to not have duplication.
Could you write API tests for it?